ImageAnalysis/image_analysis/analyzers/online_analysis_modules/image_processing_funcs.py

```diff
@@ -0,0 +1,35 @@
+"""
+3-25-2024
+
+Module to contain functions common to more than one Analyzer class, but not necessarily ubiquitous enough to include
+in the parent ImageAnalyzer class itself.
+
+-Chris
+"""
+
+import numpy as np
+
+def threshold_reduction(image, threshold):
+    return_image = np.copy(image) - threshold
+    return_image[np.where(return_image < 0)] = 0
+    return return_image
+
+
+def find_max(image):
+    y_max, x_max = np.unravel_index(np.argmax(image), image.shape)
+    max_value = image[y_max, x_max]
+    return x_max, y_max, max_value
+
+
+def saturation_check(image, saturation_value):
+    return len(np.where(image > saturation_value)[0])
+
+
+def calculate_fwhm(projection_array, threshold=None):
+    maxloc = np.argmax(projection_array)
+    halfmax = projection_array[maxloc]/2
+    if threshold is not None:
+        halfmax = halfmax + 0.5*threshold
+
+    peak_region = np.where(projection_array >= halfmax)[0]
```

Comment:
Assumes that there are no peaks elsewhere in the array.

Reply:
True.  I could fix this to account for multi-peaks, although this would be a kind of fringe case.  If our beam did have multiple spiky peaks, I would be hesitant to report that the FWHM was low for only a single peak.  Maybe there is a better term to use than FWHM, because at the end of the day if we are looking to minimize the FWHM then it is assumed we are also trying to minimize it in a single peak.